File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        index = -1
        l = 0
        r = len(nums) - 1

        while l <= r:
            mid = (l + r) // 2
            val = nums[mid]
            if val < target:
                l = mid + 1
            elif val > target:
                r = mid - 1
            else:
                return mid
        return l

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // we can use the complement approach
        HashMap<Integer, Integer> complements = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            Integer seenBeforeIndex = complements.get(complement);
            if (seenBeforeIndex != null) {
                return new int[]{seenBeforeIndex, i};
            }
            complements.put(nums[i], i);
            
        }
        return new int[]{0, 1};
    }
}
